src/ECER.Clients.Api/InviteLinks/InviteLinksEndpoints.cs

```diff
@@ -0,0 +1,34 @@
+﻿using ECER.Managers.Admin.Contract.InviteLinks;
+using ECER.Utilities.Hosting;
+using MediatR;
+using Microsoft.AspNetCore.Http.HttpResults;
+
+namespace ECER.Clients.Api.References;
+
+public class InviteLinksEndpoints : IRegisterEndpoints
+{
+  public void Register(IEndpointRouteBuilder endpointRouteBuilder)
+  {
+    endpointRouteBuilder.MapPost("/api/invitelinks", async Task<Results<Ok<GenerateInviteLinkResponse>, BadRequest<string>>> (
+      GenerateInviteLinkRequest request,
+      HttpContext httpContext,
+      IMediator messageBus,
+      IConfiguration configuration,
+      CancellationToken ct) =>
+    {
+      var referenceLinkResponse = await messageBus.Send(new GenerateInviteLinkCommand(request.portalInvitation, request.inviteType, request.validDays), ct) as GenerateInviteLinkCommandResponse;
+      string baseUrl = configuration["PortalApp:BaseUrl"]!;
+      string referenceVerificationRoute = configuration["PortalApp:ReferenceVerificationRoute"]!;
+      string verificationLink = $"{baseUrl}/{referenceVerificationRoute}/{referenceLinkResponse!.encryptedVerificationToken}";
```

Comment:
this should be moved to the engine - it is part of the transformation

Reply:
We now convert portal invites and reference types into tokens to create links in the API. In reverse, we use tokens to retrieve portal invites and reference types. If implemented in the transformation layer, we'd need to pass the entire link instead of just the token.